File: vite.config.ts
// This file is the entry point for configuring Vite 
// and its plugins using TypeScript
import { defineConfig } from 'vite'
import RubyPlugin from 'vite-plugin-ruby'

export default defineConfig({
  plugins: [RubyPlugin()],
  server: {
    // Configure the Ruby plugin in the Vite dev server
    configureServer(server) {
      // Require your Ruby configuration file
      const config = require('./path/to/your/ruby/config.rb');
      // Use the `set` method of the Vite dev server to set the `ruby` option of the Ruby plugin
      server.set('ruby', config);
    },
  },
} as any); // type casting to bypass the type error :/

File: config/vite.config.rb
module Vite
  module Config
    def self.configure(server)
      # Set the entry point of the application
      config.entry = "/app/javascript/entrypoints/application.ts"

      # Set the output directory
      config.out_dir = "dist"

      # Enable the Ruby plugin
      config.plugin_ruby.enabled = true
    end
  end
end
